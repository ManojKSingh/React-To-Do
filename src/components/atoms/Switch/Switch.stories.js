import React from 'react';
import { action } from '@storybook/addon-actions';
import {Switch} from './Switch';

export default {
  title: 'Atom/Switch',
  component: Switch,
};

export const SwitchInput = () => <Switch onChange={action('Switch Action')} />;