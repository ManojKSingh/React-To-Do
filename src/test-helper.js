import React from 'react';
import { ThemeProvider } from 'styled-components';
import renderer from 'react-test-renderer';
import themes from '../src/themes';

const theme = themes['light'];

export function renderWithTheme(component) {
  return renderer.create(
    <ThemeProvider theme={theme}>{component}</ThemeProvider>
  );
}



