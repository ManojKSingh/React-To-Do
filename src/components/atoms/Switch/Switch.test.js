import React from 'react';
import { renderWithTheme } from '../../../test-helper';
import {Switch} from './Switch';

describe('<Switch/>', () => {
  it('renders correctly', () => {
    const tree = renderWithTheme(<Switch />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});