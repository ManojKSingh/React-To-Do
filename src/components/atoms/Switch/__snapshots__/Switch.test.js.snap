// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Switch/> renders correctly 1`] = `
<label
  className="sc-AxjAm iqrGWs switch"
>
  <input
    type="checkbox"
  />
  <span
    className="slider round"
  />
</label>
`;
